XenAdmin/Wizards/HAWizard.cs

```diff
@@ -122,7 +122,13 @@ protected override void OnShown(EventArgs e)
 
         protected override void UpdateWizardContent(XenTabPage senderPage)
         {
-            if (senderPage.GetType() == typeof(AssignPriorities))
+            Type type = senderPage.GetType();

```

Comment:
This will still reset the AssignPriorities settings if the user goes all the way to the first page, but I don't think we need it as we probably want to retain the settings even if the user chooses to store the metadata on a different SR. I think it will be simpler if we don't use the IsDirty flag at all (in fact we should remove it from elsewhere at some point because it adds complexity to the wizard logic). The error here is probably the fact that the controls are populated in the PageLoadedCore overload, so we should remove this and just rename the method PopulatePageControls to PopulatePage (which is an override of the base class method and is called on FirstLoad).